### PR TITLE
Fix: #6507

### PR DESCRIPTION
**Pull request by leberus**
Fixes bad behaviour with "doo" command.

```
$ r2 -d test
Process with PID 18765 started...
= attach 18765 18765
bin.baddr 0x00400000
USING 400000
Assuming filepath /home/oscar/lab/r2/6507/test
asm.bits 64
 -- Your project name should contain an uppercase letter, 8 vowels, some numbers, and the first 5 numbers of your private bitcoin key.
[0x7f292fa9a190]> dc
Selecting and continuing: 18765
da da da
PTRACE_EVENT_EXIT pid=18765, status=0x0
[0x7f292f7a82e9]> doo popeyemarino
Wait event received by different pid 18765
Process with PID 18766 started...
File dbg:///home/oscar/lab/r2/6507/test  popeyemarino reopened in read-write mode
= attach 18766 18766
Assuming filepath /home/oscar/lab/r2/6507/test
[0x7f6b81f52190]> dc
Selecting and continuing: 18766
popeyemarino
PTRACE_EVENT_EXIT pid=18766, status=0x0
```

### CODE REVIEW

File: libr/core/file.c [hunk 1]
```diff
@@ -122,6 +122,10 @@ R_API int r_core_file_reopen(RCore *core, const char *args, int perm, int loadbi
 		// XXX - select the right backend
 		if (core->file && core->file->desc) {
 			newpid = core->file->desc->fd;
+#if __linux__
```
Comment (radare) [inline libr/core/file.c]: this shouldnt be something specific for each backend... we may have the same problem on BSD ? (doesnt seems to fail on OSX)

File: libr/debug/p/debug_native.c [hunk 2]
```diff
@@ -1117,6 +1117,12 @@ static int r_debug_native_kill (RDebug *dbg, int pid, int tid, int sig) {
 		}
 	} else {
 #endif
+#if __linux__
+	if (sig == SIGKILL && dbg->threads) {
```
Comment (radare) [inline libr/debug/p/debug_native.c]: why this is only for linux? looks pretty generic to me

File: libr/debug/p/debug_native.c [hunk 3]
```diff
@@ -1117,6 +1117,12 @@ static int r_debug_native_kill (RDebug *dbg, int pid, int tid, int sig) {
...
+		r_list_free (dbg->threads);
+		dbg->threads = NULL;
+	}
+#endif
 	if ((r_sandbox_kill (pid, sig) != -1)) ret = true;
```
Comment (radare) [inline libr/debug/p/debug_native.c]: add braces

File: libr/debug/p/debug_native.c [hunk 4]
```diff
@@ -1117,6 +1117,12 @@ static int r_debug_native_kill (RDebug *dbg, int pid, int tid, int sig) {
...
 	if (errno == 1) ret = -true; // EPERM
```
Comment (radare) [inline libr/debug/p/debug_native.c]: add braces

File: libr/core/file.c (hunk 1 above)
Comment (leberus) [inline libr/core/file.c]: main_pid is only being set in linux_debug.c, so we shouldn't have the problem on *BSD.

**Comment by radare** [comment]
yeah, but bsd also have a main pid :P im thinking that this shuold be refactored to be generic for all targets

**Comment by leberus** [comment]
actually yes, i was just refactoring linux_debug, but i think it shouldn't be a hercules effort to bring this for *BSD platforms too. Anyway,  I'd like to commit this patch at least, and in further commits I'll try to get rid of the differences about *bsd/linux regarding debugging. If you are fine with that.. :)

**Comment by radare** [comment]
Sure :)

**Pull request merged**